projects/forge-lib/src/lib/core/services/forms.service.ts

```diff
@@ -160,38 +160,39 @@ export class FormsService {
    */
   private mapFormResponse(res: any): void {
     this.form.name = res.name;
-    res.layout.components.forEach((component) => {
-
-      switch (component.type) {
-        case "Attachment": {
-          this.form.components.push(Object.assign(new AttachmentComponent(), component));
-          break;
-        }
-        case "Checkbox": {
-          this.form.components.push(Object.assign(new CheckboxComponent(), component));
-          break;
-        }
-        case "Email": {
-          this.form.components.push(Object.assign(new EmailComponent(), component));
-          break;
-        }
-        case "Number": {
-          this.form.components.push(Object.assign(new NumberComponent(), component));
-          break;
-        }
-        case "Rating": {
-          this.form.components.push(Object.assign(new RatingComponent(), component));
-          break;
+    if (res && res.layout) {
+      res.layout.components.forEach((component) => {
+        switch (component.type) {
+          case "Attachment": {
```

Comment:
We should consider making this a global enum we can reference.

Reply:
agreed, i'll add it to the backlog